lib/rex/parser/arguments.rb

```diff
@@ -70,26 +84,54 @@ def parse(args, &_block)
       def usage
         txt = ["\nOPTIONS:\n"]
 
-        fmt.sort.each do |entry|
-          fmtspec, val = entry
+        fmt.sort_by { |option| option.first.first.downcase }.each do |fmtspec, val|
           opt = val[0] ? " <opt>  " : "        "
-          txt << "    #{fmtspec.ljust(longest.length)}#{opt}#{val[1]}"
+
+          # Get all arguments for a command
+          output = fmtspec.join(', ')
+          output += opt
+
+          # Left align the fmt options and <opt> string
+          txt << "    #{(output).ljust((self.longest + opt).length)}#{val[1]}"
         end
 
         txt << ""
         txt.join("\n")
       end
 
       def include?(search)
-        fmt.include?(search)
+        fmt.keys.flatten.include?(search)
       end
 
       def arg_required?(opt)
-        fmt[opt][0] if fmt[opt]
+        value = select_value_from_fmt_option(opt)
+        return false if value.nil?
+
+        value.first
       end
 
+      def option_names
+        fmt.keys.flatten
+      end
+
+      # Return new Parser object featuring options from the base object and including the options hash that was passed in
+      def merge(to_merge = nil)
```

Comment:
```suggestion
      def merge(fmt)
```

Just confirming if there's a benefit to allowing no arguments to this function 👀  I'm not sure if we'd need to call `arguments.merge()` with no arguments